### PR TITLE
Component main

### PR DESCRIPTION
**Pull request by shmbajaj**
## Main component
- Main component consist of two components `Rich Text Editor` and `Action Bar`, 
   it is heart of note taking app.
- Rich Text Editor uses `TipTap` package and everything came from it, apart from 
  my implementation of it.
- Action Bar comprises of two components `Label List and Action Bar Icons `and 
  renders few `JSX` elements.
- Label List component consist of list of `Label` component and renders `JSX` element.
- Label component renders label name.
- Action Bar Icons components render list of actions available apart from RTE, for 
  note writing such as background color change and adding labels it.

## Some Mentions
- Menu bar of RTE is made sticky and fixed height is given to text editor.
- On each action item selection of menu bar its get highlighted.
- This component width will be fixed when layout-ed in a page, so things might look big.

## What files to check
- **src/components/***

## What to check
- UI Look and feel
- Rich Text Editor functionality

**Actual Files to check: 8**

## What files no to check
- src/components/RichTextEditor/*.js files (as they are rte-package-code)

## What not to check
- UI functionality apart from Rich Text Editor
- Responsiveness

## Preview

https://user-images.githubusercontent.com/29247011/161283441-e6b41044-0a7f-4c32-8977-61bc3a9d2792.mp4

## Deploy Preview
https://deploy-preview-2--commonplace-book.netlify.app/

### CODE REVIEW

**Review by nightlight-wander — APPROVED** [review]
1)UI looks good and clean. 
2)For some components in editor it's hard to find from symbols only what does they mean so if that package have some hover text info ,it can be added, not necessary.
3)Rich Text editor is working fine.
4) Good destructuring .
5)Couldn't get the functionality of the code as i haven't tried it and it looks like it's using many new things.

File: src/components/LabelList/LabelList.styles.module.css [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+.labelsList {
+  display: grid;
+  row-gap: 5px;
+  column-gap: 2px;
+  grid-template-rows: 1fr;
+  grid-template-columns: repeat(auto-fill, minmax(50px, 80px));
+}
+
+/* .labelsList > * {
```
Comment (SK02K1) [inline src/components/LabelList/LabelList.styles.module.css:16]: - remove commented code before merging.

File: src/components/index.js [hunk 2]
```diff
@@ -1,4 +1,3 @@
-import { Header } from "./Header";
-import { Footer } from "./Footer/Footer";
+import { Main } from "./Main/Main";
```
Comment (SK02K1) [inline src/components/index.js:1]: - here you can directly export this ```Main``` component, no need to import it.
```suggestion
export { Main } from "./Main/Main";
```

**Review by SK02K1 — APPROVED** [review]
- Learnt multiple new things while reviewing your code.
- Liked that you are using module css.

@shmbajaj I have suggested some changes that you can look into before merging.

File: src/components/LabelList/LabelList.styles.module.css (hunk 1 above)
Comment (shmbajaj) [reply src/components/LabelList/LabelList.styles.module.css:16]: Done

File: src/components/index.js (hunk 2 above)
Comment (shmbajaj) [reply src/components/index.js:1]: For thsi project I am following this approach, without any reason, sure I will use this approach in other projects.

**Pull request merged**